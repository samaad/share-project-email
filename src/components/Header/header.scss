.header {
  justify-content: space-between;
  margin-bottom: 24px;
  font-size: 1.25rem;
  font-style: normal;
  font-weight: 300;
  line-height: 27px;
  color: #050038;

  &__projectname {
    font-weight: normal;
  }

  p {
    display: flex;
  }
}
