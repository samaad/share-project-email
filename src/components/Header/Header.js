import React from "react";
import PropTypes from "prop-types";
import "./header.scss";

const BLOCK_NAME = "header";
const Header = ({ title }) => {
  return (
    <div className={BLOCK_NAME}>
      <p
        dangerouslySetInnerHTML={{
          __html: title,
        }}
      />
    </div>
  );
};

Header.propTypes = {
  title: PropTypes.string,
};

Header.defaultProps = {
  title:
    "Share&nbsp;<span class='header__projectname'>Project name</span>&nbsp;with others",
};

export default Header;
