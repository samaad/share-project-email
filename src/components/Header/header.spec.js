// Link.react.test.js
import React from "react";
import renderer from "react-test-renderer";
import Header from "./Header";
import { shallow } from "enzyme";

test("should render the header component", () => {
  const component = renderer.create(<Header />);
  expect(component).toMatchSnapshot();
});

test("should render p tag", () => {
  const component = shallow(<Header />);
  const pelment = component.find("p");
  expect(pelment).toBeTruthy();
});
