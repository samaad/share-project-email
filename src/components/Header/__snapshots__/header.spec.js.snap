// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Panel component 1`] = `
<div
  className="header"
>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "Share&nbsp;<span class='header__projectname'>Project name</span>&nbsp;with others",
      }
    }
  />
</div>
`;

exports[`should render the header component 1`] = `
<div
  className="header"
>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "Share&nbsp;<span class='header__projectname'>Project name</span>&nbsp;with others",
      }
    }
  />
</div>
`;
